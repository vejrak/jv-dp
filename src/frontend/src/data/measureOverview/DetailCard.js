// @flow

import React from 'react'
import { Button } from 'react-bootstrap'
import type { ErrorResponse, Overview, Sensor } from '../../types'
import Spinner from '../../components/Spinner'

type Props = $ReadOnly<{|
  overview: ?Overview,
  sensor: Sensor,
  showDetail: Function,
  isFetching: boolean,
  error: ?ErrorResponse,
|}>

const DetailCard = ({
  overview,
  sensor,
  showDetail,
  isFetching,
  error,
}: Props) => {
  if (isFetching) return <Spinner />
  if (error)
    return (
      <div className="alert alert-danger" role="alert">
        {error.errmsg}
      </div>
    )
  if (!overview) return null
  return (
    <div className="card mb-2 mt-2 w-25">
      <div className="card-body">
        <h5 className="card-title">Identificator: {sensor.identificator}</h5>
        <p className="card-text">Description: {sensor.description}</p>
      </div>
      <ul className="list-group list-group-flush">
        <li className="list-group-item">
          Min value: {overview.min} {overview.unit}
        </li>
        <li className="list-group-item">
          Max value: {overview.max} {overview.unit}
        </li>
        <li className="list-group-item">
          Mean value: {overview.mean} {overview.unit}
        </li>
        <li className="list-group item">
          <Button className="small m-1" onClick={() => showDetail(sensor._id)}>
            Detail
          </Button>
        </li>
      </ul>
    </div>
  )
}

export default DetailCard
