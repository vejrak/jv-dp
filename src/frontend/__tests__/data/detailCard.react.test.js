import React from 'react'
import renderer from 'react-test-renderer'
import DetailCard from '../../src/data/measureOverview/DetailCard'

const mockProps = {
  overview: { unit: 'cm', min: 0, max: 10, mean: 5 },
  sensor: { identificator: 'id1', description: 'text' },
  showDetail: () => {},
  isFetching: false,
  error: null,
}

it('renders correctly', () => {
  const tree = renderer.create(<DetailCard {...mockProps} />).toJSON()
  expect(tree).toMatchSnapshot()
})
