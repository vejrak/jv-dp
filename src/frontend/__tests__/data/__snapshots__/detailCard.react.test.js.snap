// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card mb-2 mt-2 w-25"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Identificator: 
      id1
    </h5>
    <p
      className="card-text"
    >
      Description: 
      text
    </p>
  </div>
  <ul
    className="list-group list-group-flush"
  >
    <li
      className="list-group-item"
    >
      Min value: 
      0
       
      cm
    </li>
    <li
      className="list-group-item"
    >
      Max value: 
      10
       
      cm
    </li>
    <li
      className="list-group-item"
    >
      Mean value: 
      5
       
      cm
    </li>
    <li
      className="list-group item"
    >
      <button
        className="small m-1 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Detail
      </button>
    </li>
  </ul>
</div>
`;
